src/akkudoktoreos/core/pydantic.py

```diff
@@ -51,6 +52,71 @@ def deep_update(source_dict: dict[str, Any], update_dict: dict[str, Any]) -> dic
     return merged_dict
 
 
+def access_config_value(
+    config: BaseModel, path: Union[str, Sequence[str]], setter: bool, value: Optional[Any] = None
+) -> Any:
+    """Get or set a configuration value based on the provided path.
+
+    Supports string paths (with '/' separators) or sequence paths (list/tuple).
+    Trims leading and trailing '/' from string paths.
```

Comment:
Is this specific to config? If not, you may rename to something more generic.

E.g.:
```python
def access_nested_value(
    model: BaseModel, path: Union[str, Sequence[str]], setter: bool, value: Optional[Any] = None
)
```